service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -538,6 +539,16 @@ func (d *Decoder) decodeString(s *string, v reflect.Value, fieldTag tag) error {
 	switch v.Kind() {
 	case reflect.String:
 		v.SetString(*s)
+	case reflect.Slice:
+		// To maintain backwards compatibility with the ConvertFrom family of methods
+		// which converted []byte into base64-encoded strings if the input was typed
+		if v.Type() == byteSliceType {
+			decoded, err := base64.StdEncoding.DecodeString(*s)
+			if err != nil {
+				return &UnmarshalTypeError{Value: "string", Type: v.Type()}
```

Comment:
This should also set the `Err` member of `UnmarshalTypeError`.

https://github.com/aws/aws-sdk-go/pull/1978/files#diff-7b4de96174d722928e311dc8ec32d2fbR744

Reply:
Set `Err` to the base64 decode error? I gave some thought to that but thought it *might* be confusing. This change really supports the happy path where everything lines up to support a successful upgrade. If anything isn't right, return the same error that would have been shown before if misc string data was trying to be marshalled into []byte.

I'm fine changing this as suggested, but that was my rationale behind the current design.

Reply:
Using UnmarshalError (vs. UnmarshalTypeError) works well. This output looks good:

```
--- FAIL: TestUnmarshalConvertToData (0.00s)
	shared_test.go:385: case 1, expect no error, got UnmarshalError: cannot unmarshal "string" into []uint8.
		caused by: illegal base64 data at input byte 4
```